extensions/machine-learning-services/src/modelManagement/azureModelRegistryService.ts

```diff
@@ -122,29 +123,15 @@ export class AzureModelRegistryService {
 	/**
 	 * Installs dependencies for the extension
 	 */
-	public async downloadArtifact(downloadUrl: string): Promise<string> {
-		return new Promise<string>((resolve, reject) => {
-			let msgTaskName = constants.downloadModelMsgTaskName;
-			this._apiWrapper.startBackgroundOperation({
-				displayName: msgTaskName,
-				description: msgTaskName,
-				isCancelable: false,
-				operation: async op => {
-					let tempFilePath: string = '';
-					try {
-						tempFilePath = path.join(os.tmpdir(), `ads_ml_temp_${UUID.generateUuid()}`);
-						await this._httpClient.download(downloadUrl, tempFilePath, op, this._outputChannel);
+	public async execDownloadArtifactTask(downloadUrl: string): Promise<string> {
+		let results = await utils.executeTasks(this._apiWrapper, constants.downloadModelMsgTaskName, [this.downloadArtifact(downloadUrl)], true);
+		return results[0];
```

Comment:
You don't need to do any checking here to validate that a non-empty array was returned?